server/rpc_test.go

```diff
@@ -205,3 +222,207 @@ func TestPing(t *testing.T) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Ticket.Sender, protoTicket.Sender) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Ticket.FaceValue, protoTicket.FaceValue) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Ticket.WinProb, protoTicket.WinProb) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Ticket.RecipientRandHash, protoTicket.RecipientRandHash) {
+			return false
+		}
+		if actualPayment.Ticket.SenderNonce != protoTicket.SenderNonce {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Sig, protoPayment.Sig) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Seed, protoPayment.Seed) {
+			return false
+		}
+		return true
+	})
+
+	orch.On("ProcessPayment", argMatcher, manifestID).Return(nil).Once()
+
+	err = processPayment(orch, header, manifestID)
+
+	assert.Nil(t, err)
+	orch.AssertExpectations(t)
+}
+
+func TestProcessPayment_GivenErrorFromOrch_ForwardsTheError(t *testing.T) {
+	orch := &mockOrchestrator{}
+	manifestID := core.ManifestID("some manifest")
+	protoTicket := defaultTicket(t)
+	protoPayment := defaultPayment(t, protoTicket)
+	data, err := proto.Marshal(protoPayment)
+	require.Nil(t, err)
+	header := base64.StdEncoding.EncodeToString(data)
+	orch.On("ProcessPayment", mock.Anything, mock.Anything).Return(errors.New("error from mock")).Once()
+
+	err = processPayment(orch, header, manifestID)
+
+	assert.Equal(t, "error from mock", err.Error())
+}
+
+func TestGetOrchestrator_GivenValidSig_ReturnsTranscoderURI(t *testing.T) {
+	orch := &mockOrchestrator{}
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	uri := "http://someuri.com"
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(true)
+	orch.On("ServiceURI").Return(url.Parse(uri))
+	orch.On("TicketParams", mock.Anything).Return(nil)
+
+	oInfo, err := getOrchestrator(orch, &net.OrchestratorRequest{})
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Equal(uri, oInfo.Transcoder)
+}
+
+func TestGetOrchestrator_GivenInvalidSig_ReturnsError(t *testing.T) {
+	orch := &mockOrchestrator{}
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(false)
+
+	oInfo, err := getOrchestrator(orch, &net.OrchestratorRequest{})
+
+	assert := assert.New(t)
+	assert.Contains(err.Error(), "sig")
+	assert.Nil(oInfo)
+}
+
+func TestGetOrchestrator_GivenValidSig_ReturnsOrchTicketParams(t *testing.T) {
+	orch := &mockOrchestrator{}
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	uri := "http://someuri.com"
+	expectedParams := defaultTicketParams()
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(true)
+	orch.On("ServiceURI").Return(url.Parse(uri))
+	orch.On("TicketParams", mock.Anything).Return(expectedParams)
+
+	oInfo, err := getOrchestrator(orch, &net.OrchestratorRequest{})
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Equal(expectedParams, oInfo.TicketParams)
+}
+
+type mockOrchestrator struct {
+	mock.Mock
```

Comment:
We already have a `stubOrchestrator`, could the `mock.Mock` be worked into that rather than re-defining a separate set of stubs?

Reply:
@j0sh curious to hear your take on this related [discussion](https://github.com/livepeer/go-livepeer/pull/657#discussion_r244420843) about the distinction between stubs and mocks and the possibility of having both.

Reply:
> I can see the helpers saving us some work (eg, counting method calls)

For this specific example, the `testify/mock` package seems to have a [Times()](https://godoc.org/github.com/stretchr/testify/mock#Call.Times) method that can be used to set the number of times that a mocked method should return a value and then the test would assert that the method was called the expected number of times using [AssertExpectations()](https://godoc.org/github.com/stretchr/testify/mock#Mock.AssertExpectations). I *think* this should do the trick? I haven't tested this in practice yet though.

> My initial feeling is that I'd prefer to have a uniform way of checking those conditions, but I'm open if the consensus is that the mixed approach saves us time/complexity/etc.

I personally have found the ability to set expectations prior to execution to be useful and feels cleaner than defining custom state variables in stubs. For that reason, I feel ok about a mixed approach right now.

Reply:
The ability to count invocations of a method without writing boilerplate for each occurrence is indeed nice, and I'd like it even better if that could be checked after the fact, eg `assert.Equals(1, mockObj.MethodCallCount("foo"))` . If `AssertExpectations` can give us that, all the better.

Reply:
Ah there is actually an [AssertNumberOfCalls()] method which does the trick too:

```
m.AssertNumberOfCalls(t, "Bar", 3)
```

Reply:
I suppose that works. Would be more consistent to just return the count somehow, which composes better with the existing `assert` functionality -- passing in `t` every time is ugly.